fitbenchmarking/cli/tests/test_checkpoint_handler.py

```diff
@@ -65,3 +70,340 @@ def test_create_files(self):
                     file = results / e
                     assert file.exists(), f'Failed to find "{e}"'
                     assert file.stat().st_size > 0, f'File is empty: "{e}"'
+
+
+class TestMergeDataSets(TestCase):
+    """
+    Simple testing for the checkpoint_handler merge entry point.
+    Uses the definition of identical defined in the docstring of
+    merge_data_sets
+    """
+
+    def setUp(self):
+        """
+        Generate 3 files to merge
+        """
+        ch_test_files = Path(__file__).parent / 'test_files'
+        self.A = str(ch_test_files/'A.json')
+        self.B = str(ch_test_files/'B.json')
+        self.C = str(ch_test_files/'C.json')
+        self.expected = ch_test_files / 'AB.json'
+        self._dir = TemporaryDirectory()  # pylint:disable=consider-using-with
+        self.dir = Path(self._dir.name)
+
+    def test_create_new_file(self):
+        """
+        Test that a new checkpoint file is generated.
+        """
+        output = self.dir / 'out.json'
+        merge_data_sets([self.A, self.B, self.C], str(output))
+        assert output.exists(), f'No new checkpoint file {output}'
+
+    def test_three_files(self):
+        """
+        Test that merging A, B, and C is the same as merging A and B, then
+        merging the result with C.
+        """
+        intermediate = self.dir / 'intermediate.json'
+        merge_data_sets([self.A, self.B], str(intermediate.absolute()))
+        twomerge = self.dir / 'two-merges.json'
+        merge_data_sets([str(intermediate), self.C], str(twomerge))
+        onemerge = self.dir / 'one-merge.json'
+        merge_data_sets([self.A, self.B, self.C], str(onemerge))
+
+        errors = []
+        for i, (l1, l2) in enumerate(zip(onemerge.read_text().splitlines(),
+                                         twomerge.read_text().splitlines())):
+            if l1 != l2:
+                errors.append((i, l1, l2))
+
+        assert not errors, 'Files do not match. ' \
+            f'Lines {", ".join([str(e[0]) for e in errors])} disagree.'
+
+    def test_regression_two_files(self):
+        """
+        Test that merging 2 files gives the expected output file.
+        """
+        output = self.dir / 'AB.json'
+        merge_data_sets([self.A, self.B], str(output))
+
+        errors = []
+        for i, (l1, l2) in enumerate(
+            zip(self.expected.read_text().splitlines(),
+                output.read_text().splitlines())):
+            if l1 != l2:
+                errors.append((i, l1, l2))
+
+        assert not errors, 'Files do not match. ' \
+            f'Lines {", ".join([str(e[0]) for e in errors])} disagree.'
+
+    def test_strategy_first(self):
```

Comment:
**[Optional refactoring suggestion]** These 5 functions to test strategy are very similar. Maybe they can be refactored using subtest.